### PR TITLE
Fix template for checking static urls

### LINKED ISSUE #480: file not found unless url ends with forward slash
While following along in the [Getting Started Guide](http://hanamirb.org/guides/getting-started/) I am unable to open the url `/books` but `/books/` works fine.    

For example, after running this command:

`hanami generate action web books#index`

In the browser:

  /books
    - throws error in browser: `File not found: /books`

  /books/ 
    - opens a blank page with no errors.

auto generated routes file:

```  
  get '/books', to: 'books#index'
  # Configure your routes here
  # See: http://www.rubydoc.info/gems/hanami-router/#Usage
  get '/', to: 'home#index'
```
After continuing the guide and adding the file `apps/web/templates/books/index.html.erb`

Then the `spec/web/features/list_books_spec.rb` test still fails as before, `/books` still has the same error, and `/books/` shows the new html as expected with the 2 books.

Test:

```
Run options: --seed 41756

# Running:

...E

Finished in 0.020736s, 192.8970 runs/s, 192.8970 assertions/s.

  1) Error:
list books#test_0001_lists 2 books:
Capybara::ElementNotFound: Unable to find css "#books"
    /home/laymanmu/.rbenv/versions/2.2.4/lib/ruby/gems/2.2.0/gems/capybara-2.6.0/lib/capybara/node/finders.rb:43:in `block in find'
    /home/laymanmu/.rbenv/versions/2.2.4/lib/ruby/gems/2.2.0/gems/capybara-2.6.0/lib/capybara/node/base.rb:84:in `synchronize'
    /home/laymanmu/.rbenv/versions/2.2.4/lib/ruby/gems/2.2.0/gems/capybara-2.6.0/lib/capybara/node/finders.rb:32:in `find'
    /home/laymanmu/.rbenv/versions/2.2.4/lib/ruby/gems/2.2.0/gems/capybara-2.6.0/lib/capybara/session.rb:686:in `block (2 levels) in <class:Session>'
    /home/laymanmu/.rbenv/versions/2.2.4/lib/ruby/gems/2.2.0/gems/capybara-2.6.0/lib/capybara/session.rb:287:in `within'
    /home/laymanmu/.rbenv/versions/2.2.4/lib/ruby/gems/2.2.0/gems/capybara-2.6.0/lib/capybara/dsl.rb:51:in `block (2 levels) in <module:DSL>'
    /home/laymanmu/www/bookshelf/spec/bookshelf/features/list_books_spec.rb:6:in `block (2 levels) in <top (required)>'

4 runs, 4 assertions, 0 failures, 1 errors, 0 skips
rake aborted!
Command failed with status (1): [ruby -I"lib:spec" -I"/home/laymanmu/.rbenv/versions/2.2.4/lib/ruby/gems/2.2.0/gems/rake-10.5.0/lib" "/home/laymanmu/.rbenv/versions/2.2.4/lib/ruby/gems/2.2.0/gems/rake-10.5.0/lib/rake/rake_test_loader.rb" "spec/**/*_spec.rb" ]

Tasks: TOP => test
(See full trace by running task with --trace)
```

### PR DESCRIPTION
**Pull request by davydovanton**
fixed #480

### CODE REVIEW

File: test/integration/static_assets/test_application.rb [hunk 1]
```diff
@@ -67,6 +67,22 @@
     response.status.must_equal 404
   end
 
+  it "do not block application path" do
```
Comment (cllns) [inline test/integration/static_assets/test_application.rb]: s/do/does

File: test/fixtures/static_assets_app/app/templates/home/dashboard.html.erb [hunk 2]
```diff
@@ -0,0 +1,2 @@
+dashboard
+
```
Comment (cllns) [inline test/fixtures/static_assets_app/app/templates/home/dashboard.html.erb]: Trailing new line should be removed :) 

Also maybe change the text to something like "This is a dashboard page, used to test static assets" ?

**Comment by davydovanton** [comment]
@cllns thanks for review :+1:

**Comment by hlegius** [comment]
:ok_hand:

**Comment by joneslee85** [comment]
looks :+1: . Have you checked with the bug reporters to see if it has solved the issue?

**Comment by davydovanton** [comment]
@joneslee85 yep. Without this changes I have this error:
```
~/repositories/hanami(fix-public-path-request*) » rake TEST=test/integration/static_assets/test_application.rb
Run options: --seed 40550

# Running:

...F...

Finished in 2.194916s, 3.1892 runs/s, 10.9344 assertions/s.

  1) Failure:
Serve static assets (Application)#test_0005_does not block application path [/Users/anton/work/repositories/lotus/lotus/test/integration/static_assets/test_application.rb:81]:
Expected: 200
  Actual: 404

7 runs, 24 assertions, 1 failures, 0 errors, 0 skips
```

File: lib/hanami/static.rb [hunk 3]
```diff
@@ -32,7 +32,7 @@ def call(env)
 
     def can_serve(path, original = nil)
       destination = Dir[PUBLIC_DIRECTORY].find do |file|
-        file.index(path).to_i > 0
+        file.index("public#{path}").to_i > 0
```
Comment (jodosha) [inline lib/hanami/static.rb]: @davydovanton What is `"public"`?

Comment (davydovanton) [inline lib/hanami/static.rb]: this is `public` folder. I can replace this to constant if you want.

File: test/integration/static_assets/test_application.rb [hunk 4]
```diff
@@ -81,7 +97,7 @@
 JS
       end
 
-      get '/assets/dashboard.js'
+      get 'assets/dashboard.js'
```
Comment (davydovanton) [inline test/integration/static_assets/test_application.rb]: I did it for checking problem when you write routes as like this `src="assets/application.js"`

Comment (jodosha) [inline test/integration/static_assets/test_application.rb]: @davydovanton Great :+1:

**Comment by jodosha** [comment]
@hanami/contributors Can you please manually check if this works for you? Thanks.

**Comment by SleeplessByte** [comment]
This fixed `root` reporting `File not found: /` :+1:

**Comment by Freika** [comment]
Fixed :+1:

**Comment by jodosha** [comment]
@davydovanton I've manually verified it as well both on existing and brand new project: it works well. Thank you very much :+1:

**Comment by davydovanton** [comment]
@jodosha thanks for review and helping :)

**Pull request merged**